docs/architecture/deep-dives/state/derived-state.md

```diff
@@ -0,0 +1,129 @@
+# Derived State
+
+It is common to need to cache the result of expensive work that does not represent true alterations
+in application state. Derived state exists to store this kind of data in memory and keep it up to
+date when the underlying observable state changes.
+
+## `DeriveDefinition`
```

Comment:
⛏️ and it may be too late here, but why not "derived" for the object name? This would coincide with the "derived" state lower down. Or is this a "derivation" definition?

Reply:
The thinking was just that the state has already been done, hence past tense while the definition is describing how to derive.